extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -124,8 +124,7 @@ protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm, Runn
             // handle asynchronously if not throwing
             if (!throwing && ret != null) {
                 ReactiveTypeConverter<Object> converter = null;
-                if (ret instanceof CompletionStage == false
-                        && ret instanceof Publisher == false) {
+                if (ret instanceof CompletionStage == false) {
```

Comment:
Why did you drop this clause? We don't have to convert it if it's a `Publisher`.

Reply:
We actually do.
If we don't, then the following block does not work as expected (line 145):
```
ret = handleAsync(tm, tx, ic, ret, afterEndTransaction);
// convert back
if (converter != null)
       ret = converter.fromPublisher((Publisher<?>) ret);
```

Imagine we have a `Multi`. As `Multi` is an instance of `Publisher`, you don't look up for the converter (in the current code, before this PR). However, this Publisher gets _decorated_ in `handleAsync` which returns a `Publisher`. As we don't have a converter, it is not converted back to a `Multi`... boom `ClassCastException`.  It works with RX Java because the implementation of Reactive Streams Operators (used in `handleAsync`) returned a Flowable, but if you use a Flux or a Multi it does not work.

Reply:
Ah, damn. Well, OK. Perhaps we could check the intercepted method's return type and allow reusing the `Publisher` if its return type is `Publisher` (as opposed to `Multi`)?

Reply:
That would remove the need to call the converter in this case. Something like:
```
if (ret instanceof CompletionStage == false && ! ret.getClass().equals(Publisher.class)) {
```

Reply:
Fixed.